### PR TITLE
Fixes #6976 - Build default PXE menu should be POST request

### PR DESCRIPTION
**Pull request by tstrachota**

### CODE REVIEW

File: app/controllers/api/v2/config_templates_controller.rb [hunk 1]
```diff
@@ -79,7 +79,7 @@ def destroy
         process_response @config_template.destroy
       end
 
-      api :GET, "/config_templates/build_pxe_default", N_("Update the default PXE menu on all configured TFTP servers")
+      api :POST, "/config_templates/build_pxe_default", N_("Update the default PXE menu on all configured TFTP servers")
 
       def build_pxe_default
         status, msg = ProvisioningTemplate.authorized(:deploy_provisioning_templates).build_pxe_default(self)
```
Comment (domcleal) [inline app/controllers/api/v2/config_templates_controller.rb:86]: Perhaps log a deprecation here if the request.method was GET?

**Comment by tstrachota** [comment]
@domcleal updated. I noticed that we have the same problem in provisioning templates controller that has been created recently. I fixed it there too.

File: app/controllers/api/v2/config_templates_controller.rb [hunk 2]
```diff
@@ -79,9 +79,10 @@ def destroy
         process_response @config_template.destroy
       end
 
-      api :GET, "/config_templates/build_pxe_default", N_("Update the default PXE menu on all configured TFTP servers")
+      api :POST, "/config_templates/build_pxe_default", N_("Update the default PXE menu on all configured TFTP servers")
 
       def build_pxe_default
+        ::ActiveSupport::Deprecation.warn('GET method for build pxe default is deprecated. Please use POST instead.') if request.method == "GET"
```
Comment (ShimShtein) [inline app/controllers/api/v2/config_templates_controller.rb:85]: I think it's not exactly what @domcleal  has meant, we should leave the GET variant, but before doing its work, it should log a deprecation warning.
something like this:

``` Ruby
api :GET, "/config_templates/build_pxe_default", N_("Update the default PXE menu on all configured TFTP servers")
def build_pxe_default_with_warning
  Deprecation.warn('...')
  build_pxe_default
end

api :POST, "/config_templates/build_pxe_default", N_("Update the default PXE menu on all configured TFTP servers")
def build_pxe_default
  do_work here
end
```

Comment (domcleal) [inline app/controllers/api/v2/config_templates_controller.rb:85]: Actually, it was what I meant.. I don't think we need two methods for it - checking request.method is enough, and I don't mind the apipie doc disappearing for the GET variant.  I don't think it'll cause any issue.

Comment (tstrachota) [inline app/controllers/api/v2/config_templates_controller.rb:85]: I think it's ok to document only the valid option and hide the deprecated one. Existing scripts will keep working and new users will use POST after reading the docs.

**Comment by domcleal** [comment]
Merged as 9250d8b, thanks @tstrachota.

**Pull request closed without merging**